### PR TITLE
[Blazor] Security - Server - Additional security abstractions section rework

### PR DESCRIPTION
**Pull request by hakenr**
These classes aren't abstractions in the usual sense (e.g., as used in dependency injection). In the source code, `ServerAuthenticationStateProvider` is used as the default implementation registered by `AddRazorComponents()` and `AddServerSideBlazor()`. `RevalidatingServerAuthenticationStateProvider` is a base class designed to support revalidating implementations like `IdentityRevalidatingAuthenticationStateProvider`.

I removed the paragraph explaining how to change the revalidation interval, as it was an unnecessary detail here and referenced a dead link.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/security/server/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/801002d655c94db1f03cf2189fc35f58556956e8/aspnetcore/blazor/security/server/index.md) | [aspnetcore/blazor/security/server/index](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/security/server/index?branch=pr-en-us-34056) |

### CODE REVIEW

File: aspnetcore/blazor/security/server/index.md [hunk 1]
```diff
@@ -447,19 +447,13 @@ Authentication uses the same ASP.NET Core Identity authentication as Razor Pages
...
 
-## Additional security abstractions
+## Supporting authentication state providers
 
-Two additional abstractions participate in managing authentication state:
+Two additional classes derived from <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> help with managing authentication state on server:
 
-* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): An <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to obtain authentication state from the server.
+* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): A default  <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to manage authentication state on the server, when a more specific provider is not registered.
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:454]:
```suggestion
* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): A default  <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to manage authentication state on the server, when a more specific provider isn't registered.
```

File: aspnetcore/blazor/security/server/index.md [hunk 2]
```diff
@@ -447,19 +447,13 @@ Authentication uses the same ASP.NET Core Identity authentication as Razor Pages
 
 For guidance on general state management outside of ASP.NET Core Identity, see <xref:blazor/state-management?pivots=server>.
 
-## Additional security abstractions
+## Supporting authentication state providers
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:450]:
```suggestion
## Additional authentication state providers
```

File: aspnetcore/blazor/security/server/index.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ The authentication context is only established when the app starts, which is whe
...
 
-Blazor differs from a traditional server-rendered web apps that make new HTTP requests with cookies on every page navigation. Authentication is checked during navigation events. However, cookies aren't involved. Cookies are only sent when making an HTTP request to a server, which isn't what happens when the user navigates in a Blazor app. During navigation, the user's authentication state is checked within the Blazor circuit, which you can update at any time on the server using the [`RevalidatingAuthenticationStateProvider` abstraction](#additional-security-abstractions).
+Blazor differs from a traditional server-rendered web apps that make new HTTP requests with cookies on every page navigation. Authentication is checked during navigation events. However, cookies aren't involved. Cookies are only sent when making an HTTP request to a server, which isn't what happens when the user navigates in a Blazor app. During navigation, the user's authentication state is checked within the Blazor circuit, which you can update at any time on the server using the [`RevalidatingAuthenticationStateProvider` abstraction](#supporting-authentication-state-providers).
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:23]:
```suggestion
Blazor differs from a traditional server-rendered web apps that make new HTTP requests with cookies on every page navigation. Authentication is checked during navigation events. However, cookies aren't involved. Cookies are only sent when making an HTTP request to a server, which isn't what happens when the user navigates in a Blazor app. During navigation, the user's authentication state is checked within the Blazor circuit, which you can update at any time on the server using the [`RevalidatingAuthenticationStateProvider` abstraction](#additional-authentication-state-providers).
```

File: aspnetcore/blazor/security/server/index.md [hunk 4]
```diff
@@ -447,19 +447,13 @@ Authentication uses the same ASP.NET Core Identity authentication as Razor Pages
 
 For guidance on general state management outside of ASP.NET Core Identity, see <xref:blazor/state-management?pivots=server>.
 
-## Additional security abstractions
+## Additional authentication state providers
 
-Two additional abstractions participate in managing authentication state:
+Two additional classes derived from <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> help with managing authentication state on server:
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:452]:
```suggestion
Two additional classes derived from <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> help with managing authentication state on the server:
```

File: aspnetcore/blazor/security/server/index.md [hunk 5]
```diff
@@ -447,19 +447,13 @@ Authentication uses the same ASP.NET Core Identity authentication as Razor Pages
...
 
-## Additional security abstractions
+## Additional authentication state providers
 
-Two additional abstractions participate in managing authentication state:
+Two additional classes derived from <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> help with managing authentication state on the server:
 
-* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): An <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to obtain authentication state from the server.
+* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): A default  <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to manage authentication state on the server, when a more specific provider isn't registered.
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:454]:
```suggestion
* <xref:Microsoft.AspNetCore.Components.Server.ServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Endpoints/src/DependencyInjection/ServerAuthenticationStateProvider.cs)): A default  <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> used by the Blazor framework to manage authentication state on the server when a more specific provider isn't registered.
```

File: aspnetcore/blazor/security/server/index.md [hunk 6]
```diff
@@ -447,19 +447,13 @@ Authentication uses the same ASP.NET Core Identity authentication as Razor Pages
...
-
-  The default 30 minute revalidation interval can be adjusted in [`RevalidatingIdentityAuthenticationStateProvider` (`Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs`)](https://github.com/dotnet/aspnetcore/blob/release/7.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Areas/Identity/RevalidatingIdentityAuthenticationStateProvider.cs). The following example shortens the interval to 20 minutes:
-
-  ```csharp
-  protected override TimeSpan RevalidationInterval => TimeSpan.FromMinutes(20);
-  ```
+* <xref:Microsoft.AspNetCore.Components.Server.RevalidatingServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Server/src/Circuits/RevalidatingServerAuthenticationStateProvider.cs)): A base class for <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> services that receive an authentication state from the host environment and revalidate it at regular intervals.
```
Comment (guardrex) [inline aspnetcore/blazor/security/server/index.md:456]:
```suggestion
* <xref:Microsoft.AspNetCore.Components.Server.RevalidatingServerAuthenticationStateProvider> ([reference source](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Server/src/Circuits/RevalidatingServerAuthenticationStateProvider.cs)): A base class for <xref:Microsoft.AspNetCore.Components.Authorization.AuthenticationStateProvider> services that receive an authentication state from the host environment and revalidate it at regular intervals.
```

**Comment by guardrex** [comment]
Ok, I *might* be done messing around with it. 😄

Notes ...

* We usually use contractions.
* I elevated the "when" clause to an essential clause (i.e., it's a required qualifier for the statement) by removing the comma.
* The localization team has us trying to avoid the use of gerunds (`-ing`-ending words) in section headings. They told us that they don't translate well. We do use them for common terms, such as "logging," but we avoid them when we can. I think we can go with "Additional" there.
* Let's keep a remark on the `RevalidationInterval` but roll it into the bullet point. I fixed the link for the example implementation. BTW - They dropped the `release/7.0` branch, and I have an open issue to fix the handful of 7.0 links around the repo. I'll take care of that this morning.

Look the changes over and see what you think. 👂

**Review by guardrex — APPROVED** [review]

**Review by hakenr — COMMENTED** [review]
lgtm

I'll try to keep the language-related instructions you provided in mind (like using contractions, etc.). Unfortunately, since I'm not a native speaker, you'll always need to polish my changes.

**Pull request merged**